MastodonSDK/Sources/MastodonCore/Service/API/APIService+Mute.swift

```diff
@@ -21,6 +21,36 @@ extension APIService {
         let isMuting: Bool
     }
     
+    @discardableResult
+    public func getMutes(
+        authenticationBox: MastodonAuthenticationBox
+    ) async throws -> Mastodon.Response.Content<[Mastodon.Entity.Account]> {
+        let managedObjectContext = backgroundManagedObjectContext
+        
+        let response = try await Mastodon.API.Account.mutes(
+            session: session,
+            domain: authenticationBox.domain,
+            authorization: authenticationBox.userAuthorization
+        ).singleOutput()
+        
+        let userIDs = response.value.map { $0.id }
+        let predicate = NSPredicate(format: "%K IN %@", #keyPath(MastodonUser.id), userIDs)
+        
+        let fetchRequest = MastodonUser.fetchRequest()
+        fetchRequest.predicate = predicate
+        fetchRequest.includesPropertyValues = false
+        
+        try await managedObjectContext.performChanges {
+            let accounts = try managedObjectContext.fetch(fetchRequest) as! [MastodonUser]
+            
+            for account in accounts {
+                managedObjectContext.delete(account)
```

Comment:
How would this interact with manually viewing a blocked/muted user’s profile?

Reply:
| Text | Screenshot |
|---|---|
| @j-f1 when viewing a blocked user's profile it would appear like this in the app. That's already been implemented before though.|![RocketSim_Screenshot_iPhone_14_Pro_2022-11-24_13 28 53](https://user-images.githubusercontent.com/126418/203784786-fa86aa54-58d8-437e-8b8c-12f305dd7023.png)|

Reply:
Ok, just wanted to make sure deleting the user record from the DB doesn’t have any negative consequences for this experience. (But also: would it make sense to only delete the posts from Core Data and not the actual user record?)

Reply:
🤔 Interesting… I try to block random people and enter Home Screen. Then active the app. The app just display that profile in normal way. Whatever, the good news is the app not crash.  

Because the data clean happened when app active with network requests. So it's maybe delay to some point user is skimming the data will be removed. Or the data on screen when user go back from home is marked deleted.

Reply:
I agree with @j-f1. We only needs clean the posts from home timeline (only the index, not the post itself) and notification timeline (only the index, same above). (Any others? I think we only persist these two timeline.)

So we needs find user then use reverse relationship to find the post then feed index. And please be care of the `reblogged` relationship. :)

Reply:
Another way to find the feed index is use compound predicate with relationship condition. That will be faster.

Reply:
@MainasuK I've just pushed a commit which does the data purge on mute/block action. I'm currently not sure what you mean cleaning the posts from the home timeline by index. Maybe you could, if you have some time, give me pointer or make an adjustment in this branch? This sounds like we could possibly also get rid of the CoreData model changes?